.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,69 @@
+name: E2E Test On PR
+
+on:
+  workflow_run:
+    workflows: ["Bump versions"]
+    types:
+      - completed
+  push:
+  pull_request:
+   types: [opened, synchronize, reopened, closed]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    env:
+      CI: true
+      NODE_ENV: test
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    
+    steps:
+      - name: Log node version
+        run: node --version
+
+      - uses: actions/checkout@v2
+        with:
+         fetch-depth: '0'
+      
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Run tsc
+        run: yarn tsc
+        
+      - name: Check updates in dependencies
+        run: yarn backstage-cli versions:bump
+
+      - name: Cypress run
+        uses: cypress-io/github-action@v2
+        with:
+          start: yarn dev
+          wait-on: 'http://localhost:3000'
```

Comment:
Nice, I didn't know about this `wait-on` :+1:

Reply:
@iain-b opened my eyes for this one :)